Engine_Revit_UI/Query/Layer.cs

```diff
@@ -33,66 +33,23 @@ public static partial class Query
+        static public oM.Physical.Constructions.Layer Layer(this CompoundStructureLayer compoundStructureLayer, Document doc, BuiltInCategory builtInCategory = Autodesk.Revit.DB.BuiltInCategory.INVALID, PullSettings pullSettings = null)
         {
             if (compoundStructureLayer == null)
                 return null;
 
             pullSettings = pullSettings.DefaultIfNull();
 
             oM.Physical.Constructions.Layer aLayer = new oM.Physical.Constructions.Layer();
-            oM.Physical.Materials.Material aMaterial = new oM.Physical.Materials.Material();
 
             double aThickness = compoundStructureLayer.Width;
             if (pullSettings.ConvertUnits)
                 aThickness = UnitUtils.ConvertFromInternalUnits(aThickness, DisplayUnitType.DUT_METERS);
 
             aLayer.Thickness = aThickness;
 
-            ElementId aElementId = compoundStructureLayer.MaterialId;
-            Material aMaterial_Revit = null;
-            if(aElementId != null && aElementId != Autodesk.Revit.DB.ElementId.InvalidElementId)
-                aMaterial_Revit = document.GetElement(aElementId) as Material;
-
-            if (aMaterial_Revit == null && builtInCategory != Autodesk.Revit.DB.BuiltInCategory.INVALID)
-            {
-                Category aCategory = document.Settings.Categories.get_Item(builtInCategory);
-                if (aCategory != null)
-                    aMaterial_Revit = aCategory.Material;
-            }
-
-            if (aMaterial_Revit == null)
-            {
-                Compute.MaterialNotFoundWarning(aMaterial);
-                return aLayer;
-            } 
-
-            /*switch(aMaterial_Revit.MaterialClass)
-            {
-                case "Aluminium":
-                    aMaterial.MaterialType = oM.Environment.Elements.MaterialType.Opaque;
-                    break;
-                case "Concrete":
-                    aMaterial.MaterialType = oM.Environment.Elements.MaterialType.Opaque;
-                    break;
-                case "Steel":
-                    aMaterial.MaterialType = oM.Environment.Elements.MaterialType.Opaque;
-                    break;
-                case "Metal":
-                    aMaterial.MaterialType = oM.Environment.Elements.MaterialType.Opaque;
-                    break;
-                case "Wood":
-                    aMaterial.MaterialType = oM.Environment.Elements.MaterialType.Opaque;
-                    break;
-            }*/
-
-            aMaterial.Name = aMaterial_Revit.Name;
-
-            IEnvironmentMaterial aMaterialProperties = Convert.ToBHoM(aMaterial_Revit, pullSettings) as IEnvironmentMaterial;
-
-            aMaterial.Properties.Add(aMaterialProperties);
-
-            aLayer.Material = aMaterial;
+            Autodesk.Revit.DB.Material revitMaterial = doc.GetElement(compoundStructureLayer.MaterialId) as Autodesk.Revit.DB.Material;
+            aLayer.Material = revitMaterial.ToBHoMEmptyMaterial(pullSettings);
```

Comment:
Can we set Layer name after Material name?

Reply:
Which name would you like to assign to the layer? `Autodesk.Revit.DB.CompoundStructureLayer` does not have one :<